From 359fedfb337861f8be21bd22c325dfef64b9508d Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sun, 18 Apr 2021 15:57:15 +0200
Subject: [PATCH] Fix crash on start, systemReady might be called too late to
 get mLights

Regression caused by I6094c446e0b8c23f57d30652a3cbd35dee5e821a
Fix crash on start, systemReady might be called too late to get mLights

Regression caused by I6094c446e0b8c23f57d30652a3cbd35dee5e821a

diff --git a/services/core/java/com/android/server/power/PowerManagerService.java b/services/core/java/com/android/server/power/PowerManagerService.java
index d303aeddfae..e9596e541bd 100644
--- a/services/core/java/com/android/server/power/PowerManagerService.java
+++ b/services/core/java/com/android/server/power/PowerManagerService.java
@@ -2586,9 +2586,9 @@ public final class PowerManagerService extends SystemService
                         }
                         if (getWakefulnessLocked() == WAKEFULNESS_AWAKE) {
                             if (now > mLastUserActivityTime + BUTTON_ON_DURATION) {
-                                mButtonsLight.setBrightness(PowerManager.BRIGHTNESS_OFF_FLOAT);
+                                if(mButtonsLight != null) mButtonsLight.setBrightness(PowerManager.BRIGHTNESS_OFF_FLOAT);
                             } else {
-                                mButtonsLight.setBrightness(mScreenBrightnessSettingDefault);
+                                if(mButtonsLight != null) mButtonsLight.setBrightness(mScreenBrightnessSettingDefault);
                                 nextTimeout = now + BUTTON_ON_DURATION;
                             }
                         }
@@ -2598,7 +2598,7 @@ public final class PowerManagerService extends SystemService
                         if (now < nextTimeout) {
                             mUserActivitySummary = USER_ACTIVITY_SCREEN_DIM;
                             if (getWakefulnessLocked() == WAKEFULNESS_AWAKE) {
-                                mButtonsLight.setBrightness(PowerManager.BRIGHTNESS_OFF_FLOAT);
+                                if(mButtonsLight != null) mButtonsLight.setBrightness(PowerManager.BRIGHTNESS_OFF_FLOAT);
                             }
                         }
                     }
-- 
2.25.1

